### PR TITLE
SONAR-6770 Improve logging of cache usage

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

**Comment by henryju** [comment]
LGTM

File: sonar-batch/src/main/java/org/sonar/batch/bootstrap/WSLoader.java [hunk 1]
```diff
@@ -110,71 +119,94 @@ private boolean isOffline() {
...
-        // http exceptions should always be thrown (no fallback)
-        throw serverException;
-      }
-      byte[] cached = loadFromCache(id);
-      if (cached != null) {
-        return cached;
+    } catch (NotAvailableException serverNotAvailable) {
+      try {
+        return loadFromCache(id);
+      } catch (NotAvailableException cacheNotAvailable) {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/bootstrap/WSLoader.java]: CRITICAL Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

File: sonar-batch/src/main/java/org/sonar/batch/bootstrap/WSLoader.java [hunk 2]
```diff
@@ -110,71 +119,94 @@ private boolean isOffline() {
...
 
+  @Nonnull
+  private WSLoaderResult<byte[]> loadFromCacheFirst(String id) {
     try {
-      return loadFromServer(id);
-    } catch (Exception e) {
-      if (e.getCause() instanceof HttpDownloader.HttpException) {
-        throw e;
+      return loadFromCache(id);
+    } catch (NotAvailableException cacheNotAvailable) {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/bootstrap/WSLoader.java]: CRITICAL Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

File: sonar-batch/src/main/java/org/sonar/batch/bootstrap/WSLoader.java [hunk 3]
```diff
@@ -110,71 +119,94 @@ private boolean isOffline() {
...
+      try {
+        return loadFromServer(id);
+      } catch (NotAvailableException serverNotAvailable) {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/bootstrap/WSLoader.java]: CRITICAL Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

File: sonar-batch/src/main/java/org/sonar/batch/bootstrap/WSLoader.java [hunk 4]
```diff
@@ -110,71 +119,94 @@ private boolean isOffline() {
...
       return loadFromServer(id);
-    } catch (Exception serverException) {
-      if (serverException.getCause() instanceof HttpDownloader.HttpException) {
-        // http exceptions should always be thrown (no fallback)
-        throw serverException;
-      }
-      byte[] cached = loadFromCache(id);
-      if (cached != null) {
-        return cached;
+    } catch (NotAvailableException serverNotAvailable) {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/bootstrap/WSLoader.java]: CRITICAL Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

File: sonar-batch/src/main/java/org/sonar/batch/issue/tracking/ServerIssueRepository.java [hunk 5]
```diff
@@ -104,25 +90,55 @@ public Void apply(@Nullable ServerIssue issue) {
...
+      ServerIssueConsumer consumer = new ServerIssueConsumer();
+      boolean fromCache = previousIssuesLoader.load(component.key(), consumer, true);
+      stopDebug(profiler, "Load server issues for " + component.resource().getPath(), fromCache);
+      return consumer.issueList;
     } else {
       return issuesCache.values(component.batchId());
     }
   }
 
+  private void stopDebug(Profiler profiler, String msg, boolean fromCache) {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/issue/tracking/ServerIssueRepository.java]: MAJOR Make "stopDebug" a "static" method. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* CRITICAL 4 critical
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request closed without merging**